Prevent the saving button from showing when renaming templates

youknowriad opened the pull request:
closes #48345 

## What?

This PR prevents the save button form showing in the site editor like confirmed in this issue #48345 but also prevents the "isSaving" button from showing when we trigger a direct save elsewhere (like when renaming templates).

## Why?

See reasoning here https://github.com/WordPress/gutenberg/issues/48345#issuecomment-1442190253

## Testing Instructions

1- Make an edit
2- Trigger a save from the sidebar of the site editor or using `ctrl + s`
3- The saving button is visible when the save is in progress.

1- Rename a template from the template list page
2- The saving button doesn't show up in the sidebar.

youknowriad committed:
Prevent the saving button from showing when renaming templates

Mamaduka reviewed (commented):
> 1- Rename a template from the template list page
> 2- The saving button doesn't show up in the sidebar.

@youknowriad, I think the `wasDirty` condition might need some adjustments. After saving the name change, I can still see the Save button in the sidebar.


https://user-images.githubusercontent.com/240569/221151819-89869ef2-16f3-4232-9220-beefa258b450.mp4

youknowriad committed:
Remove the wasDirty check

youknowriad commented:
It turns out my "wasDirty" logic was kind of fragile. I can't think of a way to make it work properly so basically I just reverted that part.

I see three options here:

 1- Do not show the button at all when we're "saving" changes.
 2- Accept that sometimes, the button shows up for some milliseconds while instantaneous saves like "renaming", "creating"... are triggered.
 3- Only show the "isSaving" state of the button when we actually triggered the save from the button itself (Might be a bit hard to implement properly as well)

I lean towards 1 personally

youknowriad committed:
Keep the saved changes status in the sidebar

youknowriad commented:
I've updated this PR and added the "unsaved changes" count and kept the button saying "Saved" when it should be disabled. This gets us close to the proposed designs by @jameskoster 

 - The unsaved changes count, counts the number of "entities" to be saved.
 - I'm not showing "Saved 5 minutes ago" but instead I'm showing just "Saved" for a couple reasons:
     - The first reason is that this info doesn't exist. We actually trigger multiple saves at the same time, multiple API requests, we could approximate it by keeping a "date" for the last time we triggered any "entity save" but that would have some inaccuracy (for instance when you create templates that are saved directly). Anyway, it's a challenge on its own. Another approach would be to keep a date locally in the edit-site store every time we trigger a save using the save modal (either via sidebar or within the editor or using the keyboard shortcuts)
     - The other reason is that when loading the site editor, we didn't save anything yet (should it say "0")?

I'm thinking maybe we should start simple for now. Any thoughts? @jameskoster @jasmussen

jasmussen commented:
Here's a quick GIF showing the saved state button and its behaviors:

![saved](https://user-images.githubusercontent.com/1204802/222367399-6d08f0d1-8c13-4042-a7d6-991461a7a070.gif)

I feel like this one is close to being solid, and is most definitely a big step up compared to trunk, so it's pretty key we land this one soon.

Would appreciate eyes by @richtabor and @jameskoster

jameskoster commented:
This looks great and works well, thank you.

One tiny detail I'd change; the text and button look a bit tight, could we increase the gap from 8 to 16px?

youknowriad committed:
Increase the gap between the save button and the text

ntsekouras commented:
I just cherry-picked this PR to the wp/6.2 branch to get it included in the next release: 2ac3cf1617
